samcli/commands/destroy/__init__.py

```diff
@@ -0,0 +1,309 @@
+# -*- coding: utf-8 -*-
+"""
+Init command to scaffold a project app from a template
+"""
+import logging
+import sys
+
+import boto3
+import click
+from botocore.exceptions import ClientError, WaiterError
+
+from samcli.cli.main import pass_context, common_options
+from samcli.lib.telemetry.metrics import track_command
+
+LOG = logging.getLogger(__name__)
+
+SHORT_HELP = "Destroys a deployed CloudFormation stack."
+
+HELP_TEXT = """The sam destroy command destroys a Cloudformation Stack.
+
+\b
+e.g. sam destroy -stack-name sam-app --retain-resources LOGICAL_ID_1 --retain-resources LOGICAL_ID_2
```

Comment:
I think we should accept a `--profile` argument and use it for the profile input into the boto3 client. This gives users flexibility in choosing which credentials to use.

Reply:
https://boto3.amazonaws.com/v1/documentation/api/latest/guide/configuration.html - accepting a profile will change how we construct our boto3 client.